include/irr/asset/IImageAssetHandlerBase.h

```diff
@@ -31,6 +37,77 @@ class IImageAssetHandlerBase : public virtual core::IReferenceCounted
 			return width;
 		}
 
+		static inline core::vector3du32_SIMD calcPitchInBlocks(uint32_t width, uint32_t height, uint32_t depth, uint32_t blockByteSize)
+		{
+			constexpr auto VALUE_FOR_ALIGNMENT = 1;
+			core::vector3du32_SIMD retVal;
+			retVal.X = calcPitchInBlocks(width, blockByteSize);
+			retVal.Y = calcPitchInBlocks(height, VALUE_FOR_ALIGNMENT);
+			retVal.Z = calcPitchInBlocks(depth, VALUE_FOR_ALIGNMENT);
+			return retVal;
+		}
+
+		/*
+			Patch for not supported by OpenGL R8_SRGB formats.
+			Returns converted image. Input image needs to have
+			all the regions filled and texel buffer as well.
+
+			@devshgraphicsprogramming it should support layers and mipmaps
+			as well for GLI, do we have a filter that can handle it?
+		*/
+
+		static inline core::smart_refctd_ptr<ICPUImage> convertR8ToR8G8B8Image(core::smart_refctd_ptr<ICPUImage> image)
+		{
+			core::smart_refctd_ptr<ICPUImage> newConvertedImage;
+			{
+				auto copyImageForConverting = core::smart_refctd_ptr_static_cast<ICPUImage>(image->clone());
+				auto copyImageParams = copyImageForConverting->getCreationParameters();
+				auto copyBuffer = copyImageForConverting->getBuffer();
+				auto copyRegion = copyImageForConverting->getRegions().begin();
+
+				auto newCpuBuffer = core::make_smart_refctd_ptr<ICPUBuffer>(copyBuffer->getSize());
+				memcpy(newCpuBuffer->getPointer(), copyBuffer->getPointer(), newCpuBuffer->getSize());
+
+				auto newRegions = core::make_refctd_dynamic_array<core::smart_refctd_dynamic_array<ICPUImage::SBufferCopy>>(1u);
+				ICPUImage::SBufferCopy& region = newRegions->front();
+				region.imageSubresource.mipLevel = copyRegion->imageSubresource.mipLevel;
+				region.imageSubresource.baseArrayLayer = copyRegion->imageSubresource.baseArrayLayer;
+				region.imageSubresource.layerCount = copyRegion->imageSubresource.layerCount;
+				region.bufferOffset = copyRegion->bufferOffset;
+				region.bufferRowLength = copyRegion->bufferRowLength * asset::getTexelOrBlockBytesize(EF_R8G8B8_SRGB);
+				region.bufferImageHeight = copyRegion->bufferImageHeight;
+				region.imageOffset = copyRegion->imageOffset;
+				region.imageExtent = copyRegion->imageExtent;
+
+				CMatchedSizeInOutImageFilterCommon::state_type state;
```

Comment:
you want to use `CConvertFormatImageFilter<EF_R8_SRGB, EF_R8G8B8_SRGB>::state_type`

you need to fill out a lot more state here (every single structure field)